src/main/java/io/debezium/connector/spanner/db/metadata/SchemaRegistry.java

```diff
@@ -65,16 +67,32 @@ public synchronized TableSchema getWatchedTable(TableId tableId) {
     }
 
     public synchronized void checkSchema(TableId tableId, Timestamp timestamp, List<Column> rowType) {
+        LOGGER.info("Start check schema at {}", timestamp);
         if (!validate(tableId, rowType)) {
-            if (this.updateSchema(timestamp)) {
-                schemaResetTrigger.run();
+            LOGGER.info("Schema is not validate at {}, about to start normal update", timestamp);
+            try {
+                if (this.updateSchema(timestamp)) {
+                    schemaResetTrigger.run();
+                    return;
+                }
+                else {
+                    LOGGER.info("Schema failed at {} but no exception triggered", timestamp);
+                }
+            }
+            catch (Exception e) {
```

Comment:
Make sure we only execute the logic here for exceptions that match "out of retention" errors.

So we probably need something like 

if (exception == out of retention error), update schema from stale timestamp

Reply:
Otherwise, we want to throw tje error.

Reply:
CAn we remove the try catch block here and instead add it within forceUpdateSchema? That way, only one function in this file is in charge of updating the schema.